al_aws_collector.js

```diff
@@ -175,16 +184,30 @@ class AlAwsCollector {
                         // when all else fails, stringify it the gross way with inspect
                         util.inspect(error);
             }
-            const status = this.prepareErrorStatus(errorString);
+            // post stream specific error
+            const status = streamType ? this.prepareErrorStatus(errorString, streamType, null) : this.prepareErrorStatus(errorString);
             this.sendStatus(status, () => {
                 context.fail(errorString);
             });
         } else {
             return context.succeed();
         }
     }
+    prepareHealthyStatus(streamType, streamName = 'none') {
+        return {
+            stream_name: streamName,
+            status_type: 'ok',
+            stream_type: 'status',
+            message_type: 'collector_status',
+            host_uuid: this._collectorId,
+            data: [],
+            agent_type: this._collectorType,
+            collection_type:  streamType ? streamType : this._ingestType,
+            timestamp: moment().unix()
+        };
+    }
     
-    prepareErrorStatus(errorString, streamName = 'none', collectionType, errorCode) {
+    prepareErrorStatus(errorString, collectionType, errorCode, streamName = 'none') {
```

Comment:
what is the reason for changing the function interface?
Please note this function is used in [paws-collectors](https://github.com/alertlogic/paws-collector/blob/00379a67aaf1e4569b808f5cf0c1915e089f1823/paws_collector.js#L174)

Reply:
I have updated the code and keep the interface same but for that i have added collectorType  parameter in [Paws-collector](https://github.com/alertlogic/paws-collector/pull/172/files#diff-82fd638370f7c86cddbb67ebf81bc2952ba84a1e7dc12ce1ed0fe1da30045492R173)